tests/cross_functional/system_test/test_object_expiration.py

```diff
@@ -1,96 +1,426 @@
+            ]
+        }
+
+        logger.info(f"Setting object expiration on bucket: {bucket}")
+        if version.get_semantic_ocs_version_from_config() < version.VERSION_4_11:
+            mcg_obj.s3_client.put_bucket_lifecycle_configuration(
+                Bucket=bucket, LifecycleConfiguration=expire_rule_4_10
+            )
+        else:
+            mcg_obj.s3_client.put_bucket_lifecycle_configuration(
+                Bucket=bucket, LifecycleConfiguration=expire_rule
+            )
+
+        logger.info(f"Getting object expiration configuration from bucket: {bucket}")
+        logger.info(
+            f"Got configuration: {mcg_obj.s3_client.get_bucket_lifecycle_configuration(Bucket=bucket)}"
         )
 
-    logger.info(f"Getting object expiration configuration from bucket: {bucket}")
-    logger.info(
-        f"Got configuration: {mcg_obj.s3_client.get_bucket_lifecycle_configuration(Bucket=bucket)}"
-    )
+        logger.info(f"Writing {object_key} to bucket: {bucket}")
+        assert s3_put_object(
+            s3_obj=mcg_obj, bucketname=bucket, object_key=object_key, data=obj_data
+        ), "Failed: Put Object"
+
+        expire_objects_in_bucket(bucket)
 
-    logger.info(f"Writing {object_key} to bucket: {bucket}")
-    assert s3_put_object(
-        s3_obj=mcg_obj, bucketname=bucket, object_key=object_key, data=obj_data
-    ), "Failed: Put Object"
+        logger.info(f"Getting {object_key} from bucket: {bucket} after 1 day + 8 hours")
+
+        sampler = TimeoutSampler(
+            timeout=600,
+            sleep=10,
+            func=self.check_if_objects_expired,
+            mcg_obj=mcg_obj,
+            bucket_name=bucket,
+        )
+        assert sampler.wait_for_func_status(
+            result=True
+        ), f"Objects in the bucket {bucket} are not expired"
+        logger.info("Objects in the bucket are expired as expected")
+
+    def create_obcs_apply_expire_rule(
+        self,
+        number_of_buckets,
+        cloud_providers,
+        bucket_types,
+        expiration_rule,
+        mcg_obj,
+        bucket_factory,
+    ):
+        """
+        This method will create the obcs and then apply the expire rule
+        for each obcs created
+
+        Args:
+            number_of_buckets (int): Number of buckets
+            cloud_providers (Dict): Dict representing cloudprovider config
+            bucket_types (Dict): Dict representing bucket type and respective
+                                config
+            expiration_rule (Dict): Lifecycle expiry rule
+            mcg_obj (MCG): MCG object
+            bucket_factory (Fixture): Bucket factory fixture object
+
+        Returns:
+            List: of buckets
+
+        """
+        all_buckets = create_muliple_types_provider_obcs(
+            number_of_buckets, bucket_types, cloud_providers, bucket_factory
+        )
+
+        bulk_s3_put_bucket_lifecycle_config(mcg_obj, all_buckets, expiration_rule)
+
+        return all_buckets
+
```

Comment:
can this test be run for all deployment/platforms? if not add the relevant markers where the test has to be skipped

Reply:
This test can be run on all the deployments and platforms